[Search] Update search integrations

navarone-feekery opened the pull request:
## Summary

Make the following changes to the integrations page:

- Rename `enterprise_search` category to `search`
- Remove `Application Search` tag
- Remove the JSON tile entirely

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/240770)
* Commit: b12d5d0fcda3486eb250bde33ceade0b05dc782b

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/240770#019271db-0378-40ea-a48a-9e58a5e0fc37)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/240770","number":240770,"commit":"b12d5d0fcda3486eb250bde33ceade0b05dc782b"}],"number":240770}
buildkite-pr-comment-->

sphilipse commented on x-pack/plugins/enterprise_search/server/integrations.ts:
```diff
@@ -48,7 +26,7 @@ export const registerEnterpriseSearchIntegrations = (
         defaultMessage: 'Add search to your website with the web crawler.',
       }),
-      categories: ['enterprise_search', 'app_search', 'web', 'elastic_stack', 'crawler'],
+      categories: ['app_search', 'web', 'elastic_stack', 'crawler'],
```
We're keeping the app_search category?

navarone-feekery replied:
Confirmed that we can remove it, so I've included that in this PR.

sphilipse reviewed (commented):
Instead of removing the 'enterprise_search' category, we should probably just rename it to 'search' for clarity. It's already called 'Search' in the UI, not 'Enterprise Search'. See:

src/plugins/custom_integrations/common/index.ts

navarone-feekery commented:
> Instead of removing the 'enterprise_search' category, we should probably just rename it to 'search' for clarity. It's already called 'Search' in the UI, not 'Enterprise Search'. See:

@sphilipse I've added the category back and renamed it all to `search` instead of `enterprise-search`. The sub-categories are all grouped under the new category and it reflects correctly in the URL.

Gif:
![out](https://github.com/user-attachments/assets/e82c2a3e-0b20-47f8-87f5-0b75289d60dc)

navarone-feekery commented on test/api_integration/apis/custom_integration/integrations.ts:
```diff
@@ -25,7 +25,7 @@ export default function ({ getService }: FtrProviderContext) {
         expect(resp.body).to.be.an('array');
 
-        expect(resp.body.length).to.be(56);
+        expect(resp.body.length).to.be(55);
```
This total was reduced because of the removal of the App Search JSON integration.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/241159)
* Commit: 0955c816566cdaf2178570c7e7e32c182b44d44d

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/2609a533fa950b9e7974d704fa0c3f9986816dbe) | [after](https://github.com/elastic/kibana/commit/0955c816566cdaf2178570c7e7e32c182b44d44d) | diff |
| --- | --- | --- | --- |
| `customIntegrations` | 252 | 249 | -3 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/2609a533fa950b9e7974d704fa0c3f9986816dbe) | [after](https://github.com/elastic/kibana/commit/0955c816566cdaf2178570c7e7e32c182b44d44d) | diff |
| --- | --- | --- | --- |
| `customIntegrations` | 10.3KB | 10.1KB | -169.0B |
| `searchConnectors` | 24.9KB | 24.5KB | -374.0B |
| total |  |  | -543.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/2609a533fa950b9e7974d704fa0c3f9986816dbe) | [after](https://github.com/elastic/kibana/commit/0955c816566cdaf2178570c7e7e32c182b44d44d) | diff |
| --- | --- | --- | --- |
| `customIntegrations` | 271 | 268 | -3 |


</details>


### History
* :broken_heart: [Build #241103](https://buildkite.com/elastic/kibana-pull-request/builds/241103) failed 5779222b68afaa4a062ee1b2783c144364890308
* :broken_heart: [Build #240770](https://buildkite.com/elastic/kibana-pull-request/builds/240770) failed b12d5d0fcda3486eb250bde33ceade0b05dc782b

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241159","number":241159,"commit":"0955c816566cdaf2178570c7e7e32c182b44d44d"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/241103","number":241103,"commit":"5779222b68afaa4a062ee1b2783c144364890308"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/240770","number":240770,"commit":"b12d5d0fcda3486eb250bde33ceade0b05dc782b"}],"number":241159}
buildkite-pr-comment-->

nchaulet reviewed (approved):
code LGTM 🚀
